Git Beta: commit message select crash

### Summary

<!-- Please insert a one line summary of the issue below -->

<!-- Include all steps necessary to reproduce from a clean Zed installation. Be verbose -->
Steps to trigger the problem:
1.  commit message zoom
2. input any text
3. select text

Actual Behavior:

https://github.com/user-attachments/assets/92ee7ec7-1668-4590-ae9a-51edc8227ec4

Expected Behavior:


### Zed Version and System Specs

Zed: v0.177.3 (Zed Preview) 
OS: macOS 15.3.1
Memory: 32 GiB
Architecture: aarch64

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

<details><summary>Zed.log</summary>

<!-- Click below this line and paste or drag-and-drop your log-->
```
2025-03-06T10:18:31.421056+08:00 [ERROR] connection error: error reading rpc message from socket

Caused by:
    0: WebSocket protocol error: Connection reset without closing handshake
    1: Connection reset without closing handshake
2025-03-06T10:18:31.423079+08:00 [INFO] set status on client 50393: ConnectionLost
2025-03-06T10:18:31.423877+08:00 [INFO] set status on client 50393: Reauthenticating
2025-03-06T10:18:31.424052+08:00 [INFO] set status on client 50393: Reconnecting
2025-03-06T10:18:32.143669+08:00 [INFO] connected to rpc endpoint https://collab.zed.dev/rpc
2025-03-06T10:18:33.052499+08:00 [INFO] add_connection;
2025-03-06T10:18:33.052932+08:00 [INFO] set status on client 50393: Connected { peer_id: PeerId { owner_id: 698, id: 2098833 }, connection_id: ConnectionId { owner_id: 0, id: 2 } }
2025-03-06T10:19:07.943164+08:00 [ERROR] connection error: delay between messages too long
2025-03-06T10:19:07.943471+08:00 [INFO] set status on client 50393: ConnectionLost
2025-03-06T10:19:07.943749+08:00 [INFO] set status on client 50393: Reauthenticating
2025-03-06T10:19:07.943826+08:00 [INFO] set status on client 50393: Reconnecting
2025-03-06T10:19:27.961083+08:00 [INFO] set status on client 50393: ConnectionError
2025-03-06T10:19:27.961712+08:00 [ERROR] failed to connect timed out trying to establish connection
2025-03-06T10:19:27.962064+08:00 [INFO] set status on client 50393: ReconnectionError { next_reconnection: Instant { tv_sec: 661408, tv_nsec: 956440750 } }
2025-03-06T10:19:28.467674+08:00 [INFO] set status on client 50393: Reauthenticating
2025-03-06T10:19:28.467776+08:00 [INFO] set status on client 50393: Reconnecting
2025-03-06T10:19:33.040899+08:00 [INFO] connected to rpc endpoint https://collab.zed.dev/rpc
2025-03-06T10:19:34.026968+08:00 [INFO] add_connection;
2025-03-06T10:19:34.027582+08:00 [INFO] set status on client 50393: Connected { peer_id: PeerId { owner_id: 698, id: 2099718 }, connection_id: ConnectionId { owner_id: 0, id: 3 } }
2025-03-06T10:27:05.226579+08:00 [ERROR] {
  "thread": "main",
  "payload": "There must be at least one selection",
  "location_data": {
    "file": "crates/editor/src/selections_collection.rs",
    "line": 391
  },
  "backtrace": [
    "backtrace::capture::Backtrace::create::h21b339776477734e+11629544",
    "backtrace::capture::Backtrace::new::hc8a089174c4cda25+11629352",
    "zed::reliability::init_panic_hook::{{closure}}::h1bad13577600c33d+87947868",
    "std::panicking::rust_panic_with_hook::hc9e2ec2a12b10e12+69216300",
    "std::panicking::begin_panic_handler::{{closure}}::h3b6449e2eb83bf4c+69215088",
    "std::sys::backtrace::__rust_end_short_backtrace::h38d85099ddd107a3+69209496",
    "_rust_begin_unwind+69214300",
    "core::panicking::panic_fmt::hd4003258f6ab7b29+97930832",
    "gpui::app::App::subscribe_internal::{{closure}}::h6aac8d115e5d31b6+21876136",
    "gpui::app::App::flush_effects::h5a89554a740d84c7+35253888",
    "gpui::window::Window::new::{{closure}}::h0480e116bf46c29f+35841144",
    "gpui::platform::mac::window::handle_view_event::h09a93cc2574dfb67+35664488",
    "<gpui::platform::mac::platform::MacPlatform as gpui::platform::Platform>::run::hbf73ebc78937d43e+35589556",
    "gpui::app::Application::run::hd64b4b7213c8fa5e+86349240",
    "zed::main::he6f500218819216b+87969172",
    "std::sys::backtrace::__rust_begin_short_backtrace::h97ddd24e3e5d9e0a+87817296",
    "std::rt::lang_start::{{closure}}::h67fb13fa383a43aa+89016872",
    "std::rt::lang_start_internal::h5f91760815528aa2+69150596",
    "_main+87980716"
  ],
  "app_version": "0.177.3",
  "app_commit_sha": "230972127473734b0d6806aa15c93b68ed250a0a",
  "release_channel": "preview",
  "target": "aarch64-apple-darwin",
  "os_name": "macOS",
  "os_version": "15.3.1",
  "architecture": "aarch64",
  "panicked_on": 1741228025226,
  "system_id": "d7017151-b1c8-47ba-9e3c-48e0999b1c41",
  "installation_id": "9590cfca-6a9b-4b08-aa60-0d9bc941ecd5",
  "session_id": "0193df69-9111-4d9c-a01c-1f0a37cf45cf"
}
```
<!-- Click above this line and paste or drag-and-drop your log--></details>

--- a/crates/editor/src/editor.rs
+++ b/crates/editor/src/editor.rs
@@ -2247,6 +2247,10 @@ impl Editor {
             cx.subscribe(&other, |this, other, other_evt, cx| match other_evt {
                 EditorEvent::SelectionsChanged { local: true } => {
                     let other_selections = other.read(cx).selections.disjoint.to_vec();
+                    if other_selections.is_empty() {
+                        eprintln!("no selections");
+                        return;
+                    }
                     this.selections.change_with(cx, |selections| {
                         selections.select_anchors(other_selections);
                     });
@@ -2258,6 +2262,9 @@ impl Editor {
             cx.subscribe_self::<EditorEvent>(move |this, this_evt, cx| match this_evt {
                 EditorEvent::SelectionsChanged { local: true } => {
                     let these_selections = this.selections.disjoint.to_vec();
+                    if these_selections.is_empty() {
+                        return;
+                    }
                     other.update(cx, |other_editor, cx| {
                         other_editor.selections.change_with(cx, |selections| {
                             selections.select_anchors(these_selections);

--- a/crates/editor/src/editor.rs
+++ b/crates/editor/src/editor.rs
@@ -2248,7 +2248,6 @@ impl Editor {
                 EditorEvent::SelectionsChanged { local: true } => {
                     let other_selections = other.read(cx).selections.disjoint.to_vec();
                     if other_selections.is_empty() {
-                        eprintln!("no selections");
                         return;
                     }
                     this.selections.change_with(cx, |selections| {

